portal-bridge/src/census/peers.rs

```diff
@@ -0,0 +1,103 @@
+use std::{
+    pin::Pin,
+    sync::{Arc, RwLock},
+    task::{Context, Poll},
+    time::Duration,
+};
+
+use delay_map::HashMapDelay;
+use ethportal_api::{
+    types::{
+        distance::{Distance, Metric, XorMetric},
+        portal::PongInfo,
+    },
+    Enr,
+};
+use futures::Stream;
+use rand::seq::IteratorRandom;
+use tokio::time::Instant;
+
+/// Ping delay for liveness check of peers in census
+/// One hour was chosen after 2mins was too slow, and can be adjusted
+/// in the future based on performance
+const LIVENESS_CHECK_DELAY: Duration = Duration::from_secs(3600);
+
+type PeersHashMapDelay = HashMapDelay<[u8; 32], (Enr, Distance)>;
+
+#[derive(Clone, Debug)]
+pub(super) struct Peers {
+    pub peers: Arc<RwLock<PeersHashMapDelay>>,
+}
```

Comment:
```suggestion
#[derive(Clone, Debug)]
pub(super) struct Peers {
    peers: Arc<RwLock<PeersHashMapDelay>>,
}
```

I am assuming with this paradigm this inner Peers `Arc<RwLock<PeersHashMapDelay>>` object shouldn't be pub as it being pub opens the struct to be open to deadlocks like the kbucket problem?